src/transitions.jsx

```diff
@@ -122,7 +121,7 @@ export default {
 
   },
   componentWillLeave(cb) {
-    const slide = parseInt(this.context.router.state.params.slide) || 0;
+    const slide = parseInt(this.props.currentSlide) || 0;
```

Comment:
it looks like slide is on context, so if you add `slide` to contextTypes you can reference `this.context.slide`